msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/DiscussionForumTool.java

```diff
@@ -4974,7 +4974,7 @@ public String markCheckedAsApproved()
 	  approveOrDenySelectedMsgs(true);
 	  
 	  if (numPendingMessages > 0)
-		  return PENDING_MSG_QUEUE;
+		  return null;
```

Comment:
@feralvarez95 not sure that returning null is a good idea here, does it not make sense to simply fix what is consuming this?

Reply:
I did not find a better way to solve this. I tried adding "?faces-redirect=true" to the PENDING_MSG_QUEUE and FacesContext.getCurrentInstance().getExternalContext().getFlash().setKeepMessages(true). I also tried with FacesContext.getCurrentInstance().getExternalContext().getFlash().setRedirect(true) but same result. I think it would be much easier with JSF2, so any suggestion would be great to solve this issue!